core/include/scipp/core/variable.h

```diff
@@ -339,23 +342,49 @@ class SCIPP_CORE_EXPORT Variable {
   bool operator!=(const Variable &other) const;
   bool operator!=(const VariableConstProxy &other) const;
   Variable operator-() const;
+
   Variable &operator+=(const Variable &other) &;
   Variable &operator+=(const VariableConstProxy &other) &;
-  Variable &operator+=(const double value) &;
+  template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+  Variable &operator+=(const T value) & {
+    auto v = makeVariable<T>(value);
+    v.setUnit(unit());
```

Comment:
This does not make sense? We should not silently add units.

Reply:
This is based on the comment in the previous implementation.

If we don't do this then it will only work for dimensionless Variables, so anything that has a unit will need the user to create a Variable manually first.

At this point there is an argument as to if scalar operators should really be supported as you then have multiple methods of doing the same thing.

Reply:
Well, the previous comment was a question, not a suggestion. Now that we have good unit support I think the answer should be no.

> If we don't do this then it will only work for dimensionless Variables, so anything that has a unit will need the user to create a Variable manually first.

Yes. I think that is good.

> At this point there is an argument as to if scalar operators should really be supported as you then have multiple methods of doing the same thing.

Not the same thing. One is adding a dimensionless value, the other is adding a dimensionful quantity.

Reply:
So you are saying everyone should write `var += 1.0 * sc.units.dimensionless` instead?